### PR TITLE
AB#264 Prepare endpoint for searching posts

### PR DESCRIPTION
**Pull request by steboee**
… endpoint

### CODE REVIEW

File: Controllers/RecommendationController.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            [FromQuery] PaginationRequestDTO request
+        )
+        {
+            var userAddress = GetUserAddress();
+            var user = await _userService.GetUser(userAddress);
+            var result = await _recommendationService.GetHotPosts(user, request);
+
+            // return list of all posts
+            return Ok(_mapper.Map<IEnumerable<UserPostResponseDTO>>(result));
+        }
```
Comment (brano-hozza) [inline Controllers/RecommendationController.cs:42]: Move this to post controller and remove this controller

File: Database/Neo4j/Nodes/NFTNode.cs [hunk 2]
```diff
@@ -32,6 +32,9 @@ public class NFTNode
         [JsonProperty("animationUrl")]
         public string? AnimationUrl { get; set; }
 
+        [JsonProperty("nft_vector")]
+        public string? NFTVector { get; set; }
+
```
Comment (brano-hozza) [inline Database/Neo4j/Nodes/NFTNode.cs:37]: Why we are we not using the same convention for naming_

File: Services/AuthService.cs [hunk 3]
```diff
@@ -24,10 +25,21 @@ public class AuthService : IAuthService
...
         private readonly JWTOptions _jwtOptions;
 
-        public AuthService(TokengramDbContext dbContext, IOptions<JWTOptions> jWTOptions)
+        private readonly INFTService _nftService;
+
+        private readonly IUserService _userService;
+
+        public AuthService(
+            TokengramDbContext dbContext,
+            IOptions<JWTOptions> jWTOptions,
```
Comment (brano-hozza) [inline Services/AuthService.cs:34]: rename to jwtOptions

**Review by brano-hozza — CHANGES_REQUESTED** [review]

File: Services/RecommendationService.cs [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                if (nftQueryResult == null)
+                {
+                    continue;
+                }
+
+                var postVector = nftQueryResult.NFT.NFTVector;
+                if (postVector != null)
+                {
+                    var FixedpostVector = CosineSimilarityUtil.ParseVector(postVector);
+                    var FixeduserVector = CosineSimilarityUtil.ParseVector(userPreferences);
```
Comment (kikz0rsk) [inline Services/RecommendationService.cs:88]: Use camel case for var names please

File: Services/UserService.Vector.cs [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            {
+                var keyValue = pair.Split(':');
+                if (keyValue.Length == 2 && int.TryParse(keyValue[1], out int value))
+                {
+                    vector[keyValue[0]] = value;
+                }
+            }
+
+            return vector;
+        }
```
Comment (kikz0rsk) [inline Services/UserService.Vector.cs:75]: There's already very similar function `CosineSimiliarityUtil.ParseVector`. Consider using that function and possibly add parameter for specifying separator.

**Review by kikz0rsk — CHANGES_REQUESTED** [review]

**Review by brano-hozza — APPROVED** [review]

**Review by brano-hozza — APPROVED** [review]

**Review by kikz0rsk — APPROVED** [review]

**Pull request merged**